### PR TITLE
brotli-java: initial integration

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Integrates the java code of https://github.com/google/brotli

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
@eustas - OSS-Fuzz now supports Java fuzzing, would you be interested in getting brotli-java integrated? If so, can I use your email from the CPP integration you did? https://github.com/google/oss-fuzz/pull/143

**Comment by eustas** [comment]
Sure, will work on that when I get free cycles...

File: projects/brotli-java/FuzzDecode.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+public class FuzzDecode {
+  public static void fuzzerTestOneInput(FuzzedDataProvider data) {
+    byte[] buffer = new byte[65536];
+    ByteArrayInputStream input = new ByteArrayInputStream(data.consumeBytes(65536));
+    ByteArrayOutputStream output = new ByteArrayOutputStream();
+    try {
+      BrotliInputStream brotliInput = new BrotliInputStream(input);
+      while (true) {
+        int len = brotliInput.read(buffer, 0, buffer.length);
```
Comment (eustas) [inline projects/brotli-java/FuzzDecode.java:30]: Since this is a roundtrip, would be nice to check input == output

**Comment by eustas** [comment]
(non-authoritative) LGTM

File: projects/brotli-java/FuzzDecode.java (hunk 1 above)
Comment (DavidKorczynski) [reply projects/brotli-java/FuzzDecode.java:29]: This is only a decompression test, right? i.e. the output should only be equal to the original if we compress it back, which we don't atm (the output variable was unused in fuzzer - I removed it now too so it's no longer there). I can do a roundtrip test too though, but let me do that a bit later when I too get free cycles

**Comment by DavidKorczynski** [comment]
@inferno-chromium @jonathanmetzman @oliverchang calling on the current sheriff this one is good to go

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Pull request merged**